### DISCUSSION TITLE
Terratest Execution - Dev Ops and Remote Backend

### DISCUSSION THREAD

**Discussion by ajusebastian** [gruntwork-io/knowledge-base#584] [Help]
Hi all,

We are new to the terraform and terratest concepts. we are trying to automate the infrastructure deployment for our organization using terraform through azure devops, private terraform modules in registry and remote backend. We are confused on the approach on how to use terratest to verify the plan(do a pre validation) of resources, we tried this in our local machine and was able to test a sample scenario. We do not have any clues on how to do the testing as part of devops and trigger testing whenever a merge happens in source code. Is there any recommended approach to the implementation of terratest in devops .? Is there a sample where we can refer the sample implementations. Any help on this matter is highly appreaciated.

Aju 

---

<ins datetime="2022-11-07T04:59:31Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109553">Tracked in ticket #109553</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
I think can be used as a source of inspiration existing CICD configurations in Terratest repository:
https://github.com/gruntwork-io/terratest/blob/master/.github/workflows/ci.yml#L104
https://github.com/gruntwork-io/terratest/blob/master/.circleci/config.yml#L240
